environment/environment.go

```diff
@@ -20,7 +20,8 @@ type EnvironmentInfo struct {
 	Config *EnvironmentConfig
 	State  *State
 
-	ID       string
+	ID string
+	//TODO(braa): I think we only need this for Export, now. remove and pass explicitly.
```

Comment:
Yep, AFAIK only `Export()` and `Config.Locked()` needs this.

We could remove `Config.Locked` (undocumented, unused feature) and make this explicit on `Export()`.

This would be much cleaner since now `repository` is the only thing dealing with worktrees